types/react-redux/index.d.ts

```diff
@@ -49,13 +53,72 @@ export interface InferableComponentDecorator<TOwnProps> {
  */
 export declare function connect<TOwnProps>(): InferableComponentDecorator<TOwnProps>;
 
+export declare function connect<TStateProps, TOwnProps>(
+    mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps>
+): ComponentDecorator<TStateProps, void, TOwnProps>;
+
+export declare function connect<TDispatchProps, TOwnProps>(
+    mapStateToProps: undefined,
+    mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>
+): ComponentDecorator<void, TDispatchProps, TOwnProps>;
+
 export declare function connect<TStateProps, TDispatchProps, TOwnProps>(
-    mapStateToProps?: MapStateToProps<TStateProps, TOwnProps> | MapStateToPropsFactory<TStateProps, TOwnProps>,
-    mapDispatchToProps?: MapDispatchToProps<TDispatchProps, TOwnProps> | MapDispatchToPropsFactory<TDispatchProps, TOwnProps>,
-    mergeProps?: MergeProps<TStateProps, TDispatchProps, TOwnProps>,
-    options?: Options
+    mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps>,
+    mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>
 ): ComponentDecorator<TStateProps, TDispatchProps, TOwnProps>;
 
+export declare function connect<TStateProps, TOwnProps, TMergedProps>(
```

Comment:
hmmmm, I think I'll just leave the parameters for now, and move the optionality to a separate commit